### PR TITLE
feat: CSV to JSON dataweave recipe

### PR DESCRIPTION
**Pull request by msrivastav13**
### What does this PR do?

In this recipe we show how to use DataWeave in Apex to do data transformation from CSV to JSON

### CODE REVIEW

File: force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+     */
+    public static List<Contact> convertCsvToJsonContacts(String data) {
+        DataWeave.Script script = new DataWeaveScriptResource.csvToJson();
+
+        DataWeave.Result result = script.execute(
+            new Map<String, Object>{ 'data' => data }
+        );
+        String jsonResult = result.getValueAsString();
+
+        List<Contact> contacts = (List<Contact>) JSON.deserialize(
```
Comment (pozil) [inline force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls:38]: IMO we shouldn't do this double transformation (DW + JSON parsing) as doesn't show the true value of DW: we should use DW to directly turn the CSV to `List<Contact>`.

**Comment by pozil** [comment]
BTW CI won't work for this PR as we need a pre-release org but this PR isn't directly targeting `prerelease/winter24` so it doesn't use a pre-release org.

File: force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls (hunk 1 above)
Comment (msrivastav13) [reply force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls:38]: This makes sense! Let me rework this a bit. Thank you

Comment (msrivastav13) [reply force-app/main/default/classes/DataWeaveInApex Recipes/CsvToJsonConversion.cls]: @pozil Review feedback addressed! Let me know if you see anything else.

**Review by pozil — APPROVED** [review]
Looking great, ready to merge

**Pull request merged**